hack/generated/controllers/generic_controller.go

```diff
@@ -535,3 +601,69 @@ func (gr *GenericReconciler) Patch(
 		Name:      data.metaObj.GetName(),
 	}, data.metaObj)
 }
+
+// isOwnerReady returns true if the owner is ready or if there is no owner required
+func (gr *GenericReconciler) isOwnerReady(ctx context.Context, data *ReconcileMetadata) (bool, error) {
+	_, err := gr.ResourceResolver.GetOwner(ctx, data.metaObj)
+	if err != nil {
+		switch causeErr := errors.Unwrap(err).(type) {
+		case *armresourceresolver.OwnerNotFound:
+			data.log.V(4).Info("Owner does not yet exist", "NamespacedName", causeErr.OwnerName)
+			return false, nil
+		}
+
+		return false, errors.Wrap(err, "failed to get owner")
+	}
+
+	return true, nil
+}
+
+func (gr *GenericReconciler) applyOwnership(ctx context.Context, data *ReconcileMetadata) error {
+	owner, err := gr.ResourceResolver.GetOwner(ctx, data.metaObj)
+	if err != nil {
+		return errors.Wrap(err, "failed to get owner")
+	}
+
+	if owner == nil {
+		return nil
+	}
+
+	err = gr.Patch(ctx, data, func(ctx context.Context, mutData *ReconcileMetadata) error {
+		ownerGvk := owner.GetObjectKind().GroupVersionKind()
+
+		ownerRef := metav1.OwnerReference{
+			APIVersion: strings.Join([]string{ownerGvk.Group, ownerGvk.Version}, "/"),
+			Kind:       ownerGvk.Kind,
+			Name:       owner.GetName(),
+			UID:        owner.GetUID(),
+		}
+
+		mutData.metaObj.SetOwnerReferences(ownerutil.EnsureOwnerRef(mutData.metaObj.GetOwnerReferences(), ownerRef))
+
+		mutData.log.V(4).Info("Set owner reference", "ownerGvk", ownerGvk, "ownerName", owner.GetName())
+
+		return nil
+	})
+
+	if err != nil {
+		return errors.Wrap(err, "patch owner references failed")
+	}
+
+	return nil
+}
+
+func (gr *GenericReconciler) deleteResourceSucceeded(ctx context.Context, data *ReconcileMetadata) error {
+	err := gr.Patch(ctx, data, func(ctx context.Context, mutData *ReconcileMetadata) error {
+		controllerutil.RemoveFinalizer(data.metaObj, GenericControllerFinalizer)
+		return nil
+	})
+
+	// patcher will try to fetch the object after patching, so ignore not found errors
+	if apierrors.IsNotFound(err) {
+		err = nil
+	}
+
+	data.log.V(0).Info("Deleted resource")
```

Comment:
Can/should this log message identify the resource just deleted?

Reply:
All of the `data.log` methods already have attached as key-value pairs the name and namespace of the resource in question, so the logging only needs to append whatever other information is relevant. We probably are "over-specific" in that some log messages are re-including info that's already in the log context.